tests/test_mcp.py

```diff
@@ -98,6 +102,47 @@ async def test_agent_with_stdio_server(allow_model_requests: None, openai_api_ke
         )
 
 
+async def test_server_with_duplicate_tool_names(allow_model_requests: None, openai_api_key: str):
+    class MockClient:
+        async def list_tools(self) -> types.ListToolsResult:
+            # Await the sleep coroutine to simulate an asynchronous operation
+            await asyncio.sleep(0)
+            return types.ListToolsResult(
+                tools=[
+                    types.Tool(name='x', description='', inputSchema={}),
+                    types.Tool(name='x', description='', inputSchema={}),
+                ]
+            )
+
+    class MockMCPServer(MCPServer):
```

Comment:
I think we can simplify this and likely do away with so much mocking of implementation details that may change in the future if we instead test that an MCP tool name doesn't conflict with a user-defined tool, or another MCP server's tool. We can likely use the FastMCP server we define in `tests/mcp_server.py` and already use from (the latest version of this) `test_mcp.py`